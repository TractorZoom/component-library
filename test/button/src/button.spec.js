import Button from '../../../packages/button/src';
import SaveIcon from '@material-ui/icons/Save';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Button', () => {
    it('should render with required props', () => {
        // given
        const mockData = {
            title: 'some button title',
        };

        // when
        const component = renderer.create(<Button {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render when disabled', () => {
        // given
        const mockData = {
            isDisabled: true,
            title: 'some button title',
        };

        // when
        const component = renderer.create(<Button {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render when loading', () => {
        // given
        const mockData = {
            isLoading: true,
            title: 'some button title',
        };

        // when
        const component = renderer.create(<Button {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render with custom variant', () => {
        // given
        const mockData = {
            title: 'some button title',
            variant: 'outlined',
        };

        // when
        const component = renderer.create(<Button {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render with start icon', () => {
        // given
        const mockData = {
            startIcon: <SaveIcon />,
            title: 'some button title',
        };

        // when
        const component = renderer.create(<Button {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render with start icon when loading', () => {
        // given
        const mockData = {
            isLoading: true,
            startIcon: <SaveIcon />,
            title: 'some button title',
        };

        // when
        const component = renderer.create(<Button {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });
});
