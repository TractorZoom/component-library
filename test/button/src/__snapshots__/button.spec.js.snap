// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render when disabled 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
  disabled={false}
  isDisabled={true}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="some button title"
  type="button"
>
  <span
    className="MuiButton-label"
  />
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Button should render when loading 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
  disabled={false}
  isLoading={true}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="some button title"
  type="button"
>
  <span
    className="MuiButton-label"
  />
</button>
`;

exports[`Button should render with custom variant 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="some button title"
  type="button"
>
  <span
    className="MuiButton-label"
  />
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Button should render with required props 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="some button title"
  type="button"
>
  <span
    className="MuiButton-label"
  />
</button>
`;

exports[`Button should render with start icon 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="some button title"
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
        />
      </svg>
    </span>
  </span>
</button>
`;

exports[`Button should render with start icon when loading 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
  disabled={false}
  isLoading={true}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="some button title"
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
